Leanplum-SDK/Classes/LPRequestStorage.m

```diff
@@ -34,54 +34,115 @@ - (id)init
+            [[NSFileManager defaultManager] moveItemAtPath:self.cacheFilePath
+                                                    toPath:self.documentsFilePath
+                                                     error:&error];
+            if (error) {
+                LPLog(LPVerbose, @"Error in moving stored requests: %@", error);
             }
         }
-
-        if (!count) {
-            // Backwards compatible with old cached directory
-            NSError *error;
-            if ([[NSFileManager defaultManager] fileExistsAtPath:self.cacheFilePath]) {
-                [[NSFileManager defaultManager] moveItemAtPath:self.cacheFilePath
-                                                        toPath:self.documentsFilePath
-                                                         error:&error];
-                if (error) {
-                    LPLog(LPVerbose, @"Error in moving stored requests: %@", error);
+        
+        // For compatibility with older SDKs.
+        NSMutableArray *requests = [NSMutableArray array];
+        NSInteger count = [_defaults integerForKey:LEANPLUM_DEFAULTS_COUNT_KEY];
+        if (count) {
+            for (NSInteger i = 0; i < count; i++) {
+                NSString *itemKey = [LPRequestStorage itemKeyForIndex:i];
+                NSDictionary *requestArgs = [_defaults dictionaryForKey:itemKey];
+                if (requestArgs) {
+                    [requests addObject:requestArgs];
                 }
+                [_defaults removeObjectForKey:itemKey];
             }
-            NSData *requestData = [NSData dataWithContentsOfFile:self.documentsFilePath];
-            if (requestData) {
-                error = nil;
-                NSMutableArray *savedRequests = [NSPropertyListSerialization
-                                                 propertyListWithData:requestData
-                                                 options:NSPropertyListMutableContainers
-                                                 format:NULL error:&error];
-                if (savedRequests) {
-                    requests = savedRequests;
-                }
+            [_defaults removeObjectForKey:LEANPLUM_DEFAULTS_COUNT_KEY];
+            [_defaults synchronize];
+            if (![[NSFileManager defaultManager] fileExistsAtPath:self.documentsFilePath]) {
+                [self saveRequests:requests];
             }
         }
     }
+}
 
-    LPLog(LPDebug, @"Loaded %lu requests", (unsigned long) requests.count);
+/**
+ * Save requests to documents directory as a plist format.
+ * Make sure to wrap this call with synchronize to support multi-threaded cases.
+ */
+- (void)saveRequests:(NSMutableArray *)requests
+{
+    if (requests.count > MAX_STORED_API_CALLS) {
+        NSRange range = NSMakeRange(requests.count - MAX_STORED_API_CALLS, MAX_STORED_API_CALLS);
+        requests = [[requests subarrayWithRange:range] mutableCopy];
```

Comment:
potentially faster and using less memory to remove the unwanted events?
`[requests removeObjectsAtIndexes:<#(nonnull NSIndexSet *)#>]`